astropy/time/formats.py

```diff
@@ -368,6 +369,7 @@ def value(self):
         raise NotImplementedError
 
     @classmethod
+    @functools.lru_cache(maxsize=1)
```

Comment:
Not sure what to choose for `maxsize`, but maybe it makes sense to do something a bit bigger like 32 or the default of 128? I don't think this can really chew up much memory no matter what since the output is just a subset- list of a `cls.subfmts` object references.